File: Leetcode/Easy/Find All Numbers Disappeared in an Array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int n=nums.size();
        vector<int>ans;
        for(int i=0;i<n;i++){
            while(nums[i]!=i+1 && nums[i]!=nums[nums[i]-1])swap(nums[i],nums[nums[i]-1]);
        }
        for(int i=0;i<n;i++)if(nums[i]!=i+1)ans.push_back(i+1);
        return ans;
    }
};

File: Leetcode/Easy/Find All Numbers Disappeared in an Array.java
class Solution {
    public void swap(int[] nums, int i, int j){
        int x = nums[i];
        nums[i]=nums[j];
        nums[j]=x;
    }
    
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int i=0,n=nums.length;
        while(i<n){
            while(nums[i]!=i+1){
                if(nums[nums[i]-1]==nums[i])break;
                swap(nums,i,nums[i]-1);
            }
            i++;
        }
        List<Integer> ans = new ArrayList<>();
        for(i=0;i<n;i++)
            if(nums[i]!=i+1)ans.add(i+1);
        return ans;
    }
}
